service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/ObservationMapper.java

```diff
@@ -248,6 +250,10 @@ private String mapObservationToNarrativeStatement(Observation observation, Strin
             return TemplateUtils.fillTemplate(NARRATIVE_STATEMENT_TEMPLATE, narrativeStatementTemplateParameters.build());
         }
 
+        private void markObservationAsMapped(Observation observation) {
+            idMapper.getOrNew(ResourceType.Observation, observation.getIdElement());
+        }
```

Comment:
This is how we make sure that this observation is marked as mapped, so that it doesn't end up in the output XML as a standalone statement (since it has been mapped to a nested narrative statement).

Reply:
creating dedicated method in IdMapper sounds good to me.

Reply:
This observation is already "mapped" to the CompoundStatement on line 108. Unless I'm missing something this shouldn't be needed.

Reply:
This happens only when the observation has some derived observations. If it doesn't, then:
- If it's mapped to an ObservationStatement, IdMapper will be used for generating its ID
- If it's mapped to NarrativeStatements, each of those statements will receive a new random ID, but the fact that the Observation has been mapped to something won't be registered (before this PR), as IdMapper won't be used

What we could do is generate the ID using IdMapper the same way it's done in line 108, but move that bit to the top of the method and give the variable a bit more generic name. Then, I can safely remove `markObservationAsMapped`. @mattd-kainos please let me know what you think.